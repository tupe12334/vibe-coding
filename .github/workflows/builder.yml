name: Builder CI/CD

on:
  push:
    branches:
      - main

jobs:
  lint-build-publish:
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: builder
    steps:
      - uses: actions/checkout@v4
      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 20
          registry-url: https://registry.npmjs.org
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 10.12.1
      - name: Install dependencies
        run: pnpm install
      - name: Lint
        run: pnpm run lint
      - name: Build
        run: pnpm run build
      - name: Publish
        run: pnpm publish --no-git-checks
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
