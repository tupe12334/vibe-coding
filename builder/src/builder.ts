export interface BuilderOptions {
  language: string;
}
export function builder(options: BuilderOptions) {
    
}
