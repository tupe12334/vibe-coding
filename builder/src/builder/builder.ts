import { Root } from "mdast";
import { toMarkdown } from "mdast-util-to-markdown";
import { join } from "path";
import { languageSegment } from "./language/languageSegment";

export interface BuilderOptions {
  language: string;
}
export async function builder(options: BuilderOptions) {
  const tree: Root = {
    type: "root",
    children: [
      {
        type: "heading",
        depth: 1,
        children: [{ type: "text", value: "AI agent instructions" }],
      },
    ],
  };
  const templatesPath = join(__dirname, "../../templates");

  tree.children = tree.children.concat(
    await languageSegment(templatesPath, options.language)
  );

  return toMarkdown(tree);
}
