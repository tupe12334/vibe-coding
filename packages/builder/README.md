# @vibe-code/builder

Generate a consolidated `Agents.md` file from multiple prompt templates. Use it programmatically or through the included CLI.

