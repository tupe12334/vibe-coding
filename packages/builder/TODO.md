# TODOs

1. [ ] Add test that validate that only libs projects can have release system, validate that the types are type-safe as well.

2. [x] Fix the CI/CD to publish this package to npm.
