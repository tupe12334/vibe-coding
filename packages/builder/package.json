{
  "name": "@vibe-builder/builder",
  "version": "0.1.1-6",
  "description": "Generate Vibe coding agent instructions",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vibe-code/vibe-coding.git"
  },
  "scripts": {
    "build": "tsc",
    "lint": "tsc --noEmit",
    "prebuild": "rimraf dist",
    "build:w": "tsc --watch",
    "test": "vitest run",
    "test:watch": "vitest watch",
    "test:ui": "vitest --ui",
    "test:update": "vitest run -u",
    "test:watch:update": "vitest watch -u",
    "release": "release-it"
  },
  "devDependencies": {
    "@types/mdast": "^4.0.4",
    "@types/node": "^24.0.1",
    "@vitest/ui": "3.2.3",
    "release-it": "^19.0.3",
    "typescript": "~5.1.6",
    "rimraf": "^6.0.1",
    "vitest": "^3.2.3"
  },
  "packageManager": "pnpm@10.12.1",
  "dependencies": {
    "mdast-util-to-markdown": "^2.1.2"
  },
  "publishConfig": {
    "access": "public"
  },
  "release-it": {
    "git": {
      "tagName": "v${version}",
      "commitMessage": "chore: release v${version}"
    },
    "npm": {
      "publish": true,
      "access": "public"
    },
    "github": {
      "release": true
    }
  }
}
