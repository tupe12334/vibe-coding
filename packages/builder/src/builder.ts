import { Root } from "mdast";
import { toMarkdown } from "mdast-util-to-markdown";
import { generalSegment } from "./general/generalSegment";
import { languageSegment } from "./language/languageSegment";
import { projectSegment } from "./project/projectSegment";
import { frameworkSegment } from "./framework/frameworkSegment";
import { lintSegment } from "./lint/lintSegment";
import { releaseSegment } from "./release/releaseSegment";
import { monorepoSegment } from "./monorepo/monorepoSegment";
import { cicdSegment } from "./cicd/cicdSegment";
import { BuilderOptions } from "./BuilderOptions";
import { createdAtSegment } from "./createdAt/createdAtSegment";
import { testingSegment } from "./testing/testingSegment";
import { uiSegment } from "./ui/uiSegment";

export async function builder(options?: BuilderOptions): Promise<string> {
  const tree: Root = {
    type: "root",
    children: [
      {
        type: "heading",
        depth: 1,
        children: [{ type: "text", value: "AI agent instructions" }],
      },
    ],
  };
  tree.children = tree.children.concat(await generalSegment());
  if (options === undefined) {
    return toMarkdown(tree);
  }
  const {
    language,
    packageManager,
    projectType,
    framework,
    lintSystem,
    testFramework,
    releaseSystem,
    monorepoSystem,
    cicdSystem,
    createdAt,
  } = options;
  if (packageManager) {
    tree.children.push({
      type: "paragraph",
      children: [
        {
          type: "text",
          value: `Package Manager: ${packageManager}`,
        },
      ],
    });
  }

  if (monorepoSystem) {
    tree.children = tree.children.concat(await monorepoSegment(monorepoSystem));
  }

  if (cicdSystem) {
    tree.children = tree.children.concat(await cicdSegment(cicdSystem));
  }

  if (language) {
    tree.children = tree.children.concat(await languageSegment(language));
    if (lintSystem) {
      tree.children = tree.children.concat(await lintSegment(lintSystem));
    }
  }

  if (testFramework) {
    tree.children = tree.children.concat(await testingSegment(testFramework));
  }

  if (projectType) {
    tree.children = tree.children.concat(await projectSegment(projectType));

    // Extract to function and add spec `toIncludeUIGuidelines`
    if (projectType === "frontend" || projectType === "ui-lib") {
      tree.children = tree.children.concat(await uiSegment());
    }
  }

  if (framework) {
    tree.children = tree.children.concat(await frameworkSegment(framework));
  }

  if (releaseSystem) {
    tree.children = tree.children.concat(await releaseSegment(releaseSystem));
  }

  if (createdAt) {
    tree.children = tree.children.concat(await createdAtSegment());
  }

  return toMarkdown(tree);
}
