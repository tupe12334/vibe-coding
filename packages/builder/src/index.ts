export * from "./builder";
export { BuilderOptions } from "./BuilderOptions";
export { Frameworks, getAvailableFrameworks } from "./framework";
export { Languages } from "./language";
export { ProjectTypes } from "./project";
export { LintSystems, getAvailableLintSystems } from "./lint";
export { lintSegment } from "./lint";
export { ReleaseSystems, getAvailableReleaseSystems } from "./release";
export { releaseSegment } from "./release";
export {
  MonorepoSystems,
  getAvailableMonorepoSystems,
  monorepoSegment,
} from "./monorepo";
export { CICDSystems, getAvailableCICDSystems, cicdSegment } from "./cicd";
export { createdAtSegment } from "./createdAt";
export {
  TestingFrameworks,
  getAvailableTestingFrameworks,
  testingSegment,
} from "./testing";
export { uiSegment } from "./ui/uiSegment";
export { networkMockingSegment } from "./networkMocking";
