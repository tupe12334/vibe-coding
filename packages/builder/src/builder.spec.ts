import { describe, expect, test, vi } from "vitest";
import { builder } from "./builder";
describe("builder", () => {
  test("working with no arguments returns general segment only", async () => {
    const response = await builder();

    expect(response).toMatchSnapshot();
  });

  test("working with language only with a language that is superset", async () => {
    const response = await builder({ language: "typescript" });

    expect(response).toMatchSnapshot();
  });
  test("working with language only", async () => {
    const response = await builder({ language: "javascript" });

    expect(response).toMatchSnapshot();
  });
  test("working with language and package manager", async () => {
    const response = await builder({
      language: "typescript",
      packageManager: "npm",
    });

    expect(response).toMatchSnapshot();
  });
  test("working with language and lint system", async () => {
    const response = await builder({
      language: "javascript",
      lintSystem: "eslint",
    });

    expect(response).toMatchSnapshot();
  });
  test("working with project type only", async () => {
    const response = await builder({ projectType: "frontend" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type ui-lib", async () => {
    const response = await builder({ projectType: "ui-lib" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type lib", async () => {
    const response = await builder({ projectType: "lib" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type e2e", async () => {
    const response = await builder({ projectType: "e2e" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type lib and language typescript", async () => {
    const response = await builder({
      projectType: "lib",
      language: "typescript",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with framework only - nestjs", async () => {
    const response = await builder({ framework: "nestjs" });

    expect(response).toMatchSnapshot();
  });

  test("working with framework only - react", async () => {
    const response = await builder({ framework: "react" });

    expect(response).toMatchSnapshot();
  });

  test("working with framework only - playwright", async () => {
    const response = await builder({ framework: "playwright" });

    expect(response).toMatchSnapshot();
  });

  test("working with framework only - cypress", async () => {
    const response = await builder({ framework: "cypress" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and framework - backend with nestjs", async () => {
    const response = await builder({
      projectType: "backend",
      framework: "nestjs",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and framework - frontend with react", async () => {
    const response = await builder({
      projectType: "frontend",
      framework: "react",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and framework - e2e with playwright", async () => {
    const response = await builder({
      projectType: "e2e",
      framework: "playwright",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with language, project type and framework", async () => {
    const response = await builder({
      language: "typescript",
      projectType: "backend",
      framework: "nestjs",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with release system only", async () => {
    const response = await builder({ releaseSystem: "release-it" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and release system", async () => {
    const response = await builder({
      projectType: "lib",
      releaseSystem: "semantic-release",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with monorepo system only", async () => {
    const response = await builder({ monorepoSystem: "turbo" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and monorepo system", async () => {
    const response = await builder({
      projectType: "lib",
      monorepoSystem: "nx",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with cicd system only", async () => {
    const response = await builder({ cicdSystem: "github-actions" });

    expect(response).toMatchSnapshot();
  });

  test("working with test framework only", async () => {
    const response = await builder({ testFramework: "jest" });

    expect(response).toMatchSnapshot();
  });

  test("working with project type and cicd system", async () => {
    const response = await builder({
      projectType: "lib",
      cicdSystem: "gitlab-ci",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with all options including framework", async () => {
    const response = await builder({
      language: "typescript",
      packageManager: "npm",
      lintSystem: "eslint",
      projectType: "frontend",
      framework: "react",
    });

    expect(response).toMatchSnapshot();
  });

  test("working with createdAt option", async () => {
    const date = new Date("2020-01-01T00:00:00.000Z");
    vi.useFakeTimers();
    vi.setSystemTime(date);
    const response = await builder({ createdAt: true });
    expect(response).toMatchSnapshot();
    vi.useRealTimers();
  });
});
