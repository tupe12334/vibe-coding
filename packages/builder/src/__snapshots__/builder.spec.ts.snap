// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`builder > working with all options including framework 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

Package Manager: npm

## Programming language (typescript)

* Use Interfaces over Types unless needed.

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.

## Lint system (eslint)

* Use ESLint to enforce code quality and style guidelines.

* Fix lint issues before committing code.

## Project type (frontend)

* Prefer working with flex layouts for responsive design.

* Use rems for font sizes and spacing to ensure scalability.

* When writing a component, try to enter logic or network into it.

## Framework (react)

* Use key in React lists to help React identify which items have changed, are added, or removed.
"
`;

exports[`builder > working with framework only - nestjs 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Framework (nestjs)

* Separate the folder structure into domain driven design (DDD) modules, i.e. each module should have its own folder with controllers, services, and entities.
"
`;

exports[`builder > working with framework only - react 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Framework (react)

* Use key in React lists to help React identify which items have changed, are added, or removed.
"
`;

exports[`builder > working with language and lint system 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Programming language (javascript)

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.

## Lint system (eslint)

* Use ESLint to enforce code quality and style guidelines.

* Fix lint issues before committing code.
"
`;

exports[`builder > working with language and package manager 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

Package Manager: npm

## Programming language (typescript)

* Use Interfaces over Types unless needed.

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.
"
`;

exports[`builder > working with language only 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Programming language (javascript)

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.
"
`;

exports[`builder > working with language only with a language that is superset 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Programming language (typescript)

* Use Interfaces over Types unless needed.

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.
"
`;

exports[`builder > working with language, project type and framework 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Programming language (typescript)

* Use Interfaces over Types unless needed.

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.

## Project type (backend)

* Always validate endpoint request data.

## Framework (nestjs)

* Separate the folder structure into domain driven design (DDD) modules, i.e. each module should have its own folder with controllers, services, and entities.
"
`;

exports[`builder > working with no arguments returns general segment only 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them
"
`;

exports[`builder > working with project type and framework - backend with nestjs 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Project type (backend)

* Always validate endpoint request data.

## Framework (nestjs)

* Separate the folder structure into domain driven design (DDD) modules, i.e. each module should have its own folder with controllers, services, and entities.
"
`;

exports[`builder > working with project type and framework - frontend with react 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Project type (frontend)

* Prefer working with flex layouts for responsive design.

* Use rems for font sizes and spacing to ensure scalability.

* When writing a component, try to enter logic or network into it.

## Framework (react)

* Use key in React lists to help React identify which items have changed, are added, or removed.
"
`;

exports[`builder > working with project type lib 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Project type (lib)

* Work with version control for publish the package if its publishable

* Create a clear and comprehensive README.md with installation instructions, usage examples, and API documentation

* Use semantic versioning (semver) for package versions

* Include proper TypeScript declaration files (.d.ts) for better developer experience

* Set up automated testing with good test coverage before publishing

* Configure proper entry points in package.json (main, module, types fields)

* Consider tree-shaking compatibility by using ES modules

* Add proper keywords and description in package.json for discoverability
"
`;

exports[`builder > working with project type lib and language typescript 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Programming language (typescript)

* Use Interfaces over Types unless needed.

* Use \\\`const\\\` over \\\`let\\\` unless reassignment is needed.

* Use \\\`===\\\` over \\\`==\\\` for strict equality checks.

* Use arrow functions for anonymous functions.

* Use template literals for string interpolation.

## Project type (lib)

* Work with version control for publish the package if its publishable

* Create a clear and comprehensive README.md with installation instructions, usage examples, and API documentation

* Use semantic versioning (semver) for package versions

* Include proper TypeScript declaration files (.d.ts) for better developer experience

* Set up automated testing with good test coverage before publishing

* Configure proper entry points in package.json (main, module, types fields)

* Consider tree-shaking compatibility by using ES modules

* Add proper keywords and description in package.json for discoverability
"
`;

exports[`builder > working with project type only 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Project type (frontend)

* Prefer working with flex layouts for responsive design.

* Use rems for font sizes and spacing to ensure scalability.

* When writing a component, try to enter logic or network into it.
"
`;

exports[`builder > working with project type ui-lib 1`] = `
"# AI agent instructions

## General Guidelines

* Don't work on any of the tasks in the TODO.md file unless you are asked to.

* Always try to early return from functions.

* Use \\\`const\\\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

* Write predictable functions and make a spec file for them according to the testing library in use.

* Don't use \\\`any\\\` in TypeScript, use \\\`unknown\\\` instead.

* Don't cast types without validation.

* Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them

## Project type (ui-lib)

* Prefer working with flex

* Prioritize the use of components that exist in the library already.

* Folder structure:

  \\- \\\`src/\\\`: Contains the source code for the components.
  \\- \\\`src/\\<component>/\\\`: Contains the individual components.
  &#x20; \\- Each component has its own folder with the following structure:
  &#x20;   \\- \\\`index.ts\\\`: Exports all the relevant data.
  &#x20;   \\- \\\`\\<ComponentName>.tsx\\\`: Contains the main component code.
  &#x20;   \\- \\\`styles.ts\\\`: Contains styles for the component.
  &#x20;   \\- \\\`types.ts\\\`: Contains TypeScript types for the component.
  &#x20;   \\- \\\`\\<ComponentName>.test.tsx\\\`: Contains tests for the component.
  &#x20;   \\- \\\`\\<ComponentName>.stories.tsx\\\`: Contains Storybook stories for the component.
  \\- \\\`src/index.ts\\\`: Exports all components for easy import.
"
`;
